docs/source/integrations/working_with_sql.md

```diff
@@ -0,0 +1,190 @@
+  - Pluggable materialization (table/view/CTE)
+  - No JVM/Spark overhead
+  - In-memory/mock backends for testing
+- **Pros**
+
+  - Consistent API everywhere
+  - Retains engine optimizations
+  - Excellent testability & reproducibility
+- **Cons**
+
+  - Some backends may miss niche SQL operations
+  - Migration effort for legacy pipelines may not justify moving functional code, prioritise new pipelines / projects
+
+```{warning}
+Consult the [Ibis support matrix](https://ibis-project.org/backends/support/matrix) to verify that needed SQL functions are available. You can still write raw SQL for missing features.
+```
+
+### Recommended Ibis development workflow
+
+<details>
+<summary>Iterate locally for instant, low cost feedback. Deploy to production your warehouse for scale—ensuring consistency, testability, and zero drift between environments.</summary>
+
+#### DuckDB (or SQLlite) for Development & [CI](https://en.wikipedia.org/wiki/Continuous_integration)
+
+- In-memory, zero-config engine
+- Fast startup for local iteration and unit tests
+   ```yaml
+   # conf/base/catalog.yml
+   orders:
+     type: ibis.TableDataset
+     backend: duckdb
+     filepath: data/03_primary/dev.duckdb
+     table: orders
+   ```
+
+#### Warehouse Backend for Production & [CD](https://en.wikipedia.org/wiki/Continuous_delivery) (for example Snowflake or BigQuery)
+
+- Simply swap your catalog’s backend and credentials
+- Same Ibis expressions compile to your warehouse’s SQL
+- Leverage scale, governance, cost controls
+   ```yaml
+   # conf/prod/catalog.yml
+   orders:
+     type: ibis.TableDataset
+     backend: snowflake         # or bigquery and more.
+     credentials: warehouse_creds
+     table: analytics.orders
+   ```
+
+Example pattern:
+
+<svg xmlns="http://www.w3.org/2000/svg"
+     viewBox="0 0 800 600" width="800" height="600">
+  <image href="https://kroki.io/mermaid/svg/eNqNUk1PwjAYvu9XvNnByGGQePRgwhgSooaJ4MUYU7p3o7G0S7uhJPx4-zHYUEi8be3zPl99cy6_6JqoChZJAKDrVaFIuYZwwiqIFRF0jTo0NwDJ-PUtzIm-zUlEZYbRyl1Dhlvwn-G7A6bzWXIWuSFMdKEosuBEdGTAK6LRC45myfjIUyBRGh4wUxJSViJnAsHiu0zGIkTRnZsMvJH2_1SJVITLookWD19apZxxdLaBSpEPrJ8B9fD-bsM7IT9G95NLY6WS2Z-xxqa1Y33tw1qbdvdO_9z5QeTU-_gbaV0xKQzflikpNiiqprLp0Q8nZSVL7-hRUsLtQyGXpUXDwEDhGvtFH5KafiZxr5trfmTRqLaoIDVpavpbsyGIWfFco9r1uhltJN_9NGjbcideInBNTH1es4AsJ7SyoWfLRbpctAtktzNyzz0UhO8003AFTyYW107Q013iufkn0Q9es-6f"
+         width="100%" height="100%"
+         preserveAspectRatio="xMidYMid meet" />
+</svg>
+
+<!-- This is the base64 version of the mermaid diagram that can be decoded and edited in the mermaid.live editor -->
+
+
+
+</details>
+
+## Limitations and key design decisions
+
+More broadly, there are some wider limitations to be aware of when working with Kedro & SQL:
+
+- **No conditional branching**
+Kedro does not support conditional nodes, making UPSERT logic difficult. Kedro favors reproducibility with append & overwrite modes over in-place updates/deletes.
```

Comment:
This confuses me. How is "overwrite" so different from "update"? Both "break lineage" (see table at the beginning of the page)